src/plugins/event-plugins/JsErrorPlugin.ts

```diff
@@ -73,7 +73,10 @@ export class JsErrorPlugin implements Plugin {
         const errorEvent: JSErrorEvent = {
             version: '1.0.0',
             type: event.type,
-            message: event.reason
+            message:
+                typeof event.reason === 'string'
+                    ? event.reason
+                    : JSON.stringify(event.reason)
         };
         this.recordEvent(JS_ERROR_EVENT_TYPE, errorEvent);
```

Comment:
In the case where `event.reason` is an object, it likely inherits from `Error`. In this case, we should extract type, message, file, line number, stack trace, etc.

This can be accomplished with the proposed change.
```suggestion
        this.record(event.reason);
```

Reply:
Actually, the above omits `event.type`, might need to think through a little more.

Reply:
Will investigate a bit further for this bug fix and raise a separate PR once ready.